compiler/src/dotty/tools/dotc/core/Scopes.scala

```diff
@@ -159,19 +159,27 @@ object Scopes {
     }
 
     /** The scope that keeps only those symbols from this scope that match the
-     *  given predicates. If all symbols match, returns the scope itself, otherwise
-     *  a copy with the matching symbols.
+     *  given predicates, renamed with the given rename function.
+     *  If all symbols match and none are renamed, returns the scope itself, otherwise
+     *  a copy with the matching and renamed symbols.
      */
-    final def filteredScope(p: Symbol => Boolean)(using Context): Scope = {
+    final def filteredScope(
+        keep: Symbol => Boolean,
+        rename: (Symbol, Name) => Name = (_, name) => name)(using Context): Scope =
       var result: MutableScope | Null = null
-      for (sym <- iterator)
-        if (!p(sym)) {
-          if (result == null) result = cloneScope
+      for sym <- iterator do
+        def drop() =
+          if result == null then result = cloneScope
           result.nn.unlink(sym)
-        }
+        if keep(sym) then
+          val newName = rename(sym, sym.name)
+          if newName ne sym.name then
```

Comment:
This PR fails probably because `ne` rather than `!=` is used here?

Reply:
I doubt it, but maybe you can try?

Reply:
sure, I will take a look tomorrow

Reply:
You can edit it directly on phone:)

Reply:
Indeed. Let me try that.